client/components/happychat/composer.jsx

```diff
@@ -37,23 +32,47 @@ export const Composer = createReactClass( {
 		disabled: PropTypes.bool,
 		message: PropTypes.string,
 		onFocus: PropTypes.func,
-		onSendChatMessage: PropTypes.func,
-		onUpdateChatMessage: PropTypes.func,
+		onSendMessage: PropTypes.func,
+		onSendTyping: PropTypes.func,
+		onSendNotTyping: PropTypes.func,
+		onSetCurrentMessage: PropTypes.func,
 		translate: PropTypes.func, // localize HOC
 	},
 
+	onChange( event ) {
+		const { onSendTyping, onSendNotTyping, onSetCurrentMessage } = this.props;
+
+		const sendThrottledTyping = throttle(
```

Comment:
@nosolosw This isn't actually throttling anything. The `onChange` event is called every time a character is typed, and since you're re-creating this function and the throttle every time `onChange` is called, the throttled function is different every time so it never actually throttles anything.

Unless you're taking advantage of closures by referencing outer-function variables in the inner function (which this doesn't) it's very rare you'll ever want to define a function inside another function in JavaScript in this way.

Reply:
Fixed at https://github.com/Automattic/wp-calypso/pull/19420